### PR TITLE
Add KeyDevs tool

### PR DESCRIPTION
**Pull request by smeng3408**

### CODE REVIEW

File: kfinance/domains/key_developments/key_devs_models.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+class KeyDevelopment(BaseModel):
+    """A single key development event."""
+
+    key_dev_id: int
+    situation: str
+    announced_date_utc: datetime
+    most_important_date_utc: datetime
+    source: str
+    company_role: str
```
Comment (srisi-kensho) [inline kfinance/domains/key_developments/key_devs_models.py]: the backend has a lot of these as potentially none: 
```

    key_dev_id: int
    key_dev_type: str
    situation: str | None = None
    announced_date_utc: datetime | None = None
    most_important_date_utc: datetime | None = None
    company_role: str | None = None
    source: str | None = None
```
which is correct?

File: kfinance/domains/key_developments/tests/test_key_devs_tools.py [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+                    }
+                ],
+                "Earnings Releases": [
+                    {
+                        "key_dev_id": 1002,
+                        "situation": "S&P Global reports Q1 2025 earnings",
+                        "announced_date_utc": "2025-04-29T12:00:00Z",
+                        "most_important_date_utc": "2025-04-29T12:00:00Z",
+                        "source": "Company Report",
+                        "company_role": "Subject",
```
Comment (srisi-kensho) [inline kfinance/domains/key_developments/tests/test_key_devs_tools.py:46]: if some of the fields are optional, can you include a test example where they are None?

**Review by srisi-kensho — CHANGES_REQUESTED** [review]

File: kfinance/domains/key_developments/key_devs_models.py (hunk 1 above)
Comment (smeng3408) [reply kfinance/domains/key_developments/key_devs_models.py]: the backend, updating this now.

**Review by srisi-kensho — APPROVED** [review]
